<launch>
    <!-- Set use_sim_time if playing from a bag file -->
    <arg name="use_sim_time" default="true"/>
    
    <param name="/use_sim_time" value="$(arg use_sim_time)"/>
    <param name="image_topic" value="/camera/rgb/image_raw"/>
    <param name="edge_topic" value="/edge_detection/output"/>

    <!-- Start Edge Detection Node -->
    <node name="edge_detection_node" pkg="edge_detection" type="edge_detection_node.py" output="screen">
        <param name="image_topic" value="/camera/rgb/image_raw"/>
        <param name="edge_topic" value="/edge_detection/output"/>
    </node>

    <!-- Start RViz with custom configuration -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find edge_detection)/config/edge_detection.rviz" />

      <!-- Edge to 3D Conversion Node -->
    <node name="edge_to_3d_node" pkg="edge_detection" type="edge_to_3d_node.py" output="screen" />

    
    <node pkg="tf" type="static_transform_publisher" name="root_to_camera"
        args="0 0 0 0 0 0 1 root_link camera_link 10"/>



</launch>
